rewriter/src/main/java/com/facebook/coresql/rewriter/OrderByRewriter.java

```diff
@@ -35,31 +35,30 @@
 public class OrderByRewriter
         extends Rewriter
 {
-    PatternMatcher<Set<AstNode>> matcher;
     public Set<AstNode> patternMatchedNodes;
     private static final String REWRITE_NAME = "ORDER BY without LIMIT";
 
     public OrderByRewriter()
     {
-        this.matcher = new OrderByPatternMatcher();
         this.patternMatchedNodes = new HashSet<>();
     }
 
     @Override
-    public boolean rewritePatternIsPresent(String sql)
+    public boolean hasRewritePattern(AstNode root)
     {
-        AstNode root = requireNonNull(parseStatement(sql));
-        patternMatchedNodes = matcher.matchPattern(root);
+        requireNonNull(root, "AST passed to rewriter was null");
+        patternMatchedNodes = new OrderByPatternMatcher().matchPattern(root);
         return !patternMatchedNodes.isEmpty();
     }
 
     @Override
-    public RewriteResult rewrite(String originalSql)
+    public Optional<RewriteResult> rewrite(AstNode root)
     {
-        AstNode root = requireNonNull(parseStatement(originalSql));
-        patternMatchedNodes = matcher.matchPattern(root);
+        if (!hasRewritePattern(root)) {
+            return Optional.empty();
+        }
         String rewrittenSql = Unparser.unparse(root, this);
-        return new RewriteResult(REWRITE_NAME, originalSql, rewrittenSql);
+        return Optional.of(new RewriteResult(REWRITE_NAME, Unparser.unparse(root), rewrittenSql));
```

Comment:
Remove the from string in the result. We don't need that because it's from the original input or the previous rewrite. So when we have the driver it's obvious. We don't need the extra work.

Reply:
Done